webui/react/src/components/Navigation.tsx

```diff
@@ -39,10 +40,11 @@ const Navigation: React.FC<Props> = ({ children }) => {
     return () => canceler.abort();
   }, [canceler, fetchResourcePools]);
 
+  const makeListRoleCalls = useFeature().isOn('rbac');
   useEffect(() => {
-    fetchKnownRoles();
+    makeListRoleCalls && fetchKnownRoles();
```

Comment:
You should really use an `if` statement here, `&&` as a guard is used in react code because it's expression-based.